src/com/google/enterprise/adaptor/AdaptorContext.java

```diff
@@ -14,13 +14,20 @@
 
 package com.google.enterprise.adaptor;
 
+import com.google.enterprise.adaptor.testing.UnsupportedAdaptorContext;
+
 import com.sun.net.httpserver.HttpContext;
 import com.sun.net.httpserver.HttpExchange;
 import com.sun.net.httpserver.HttpHandler;
 
 /**
  * Methods for an Adaptor to communicate with the adaptor library.
  * Implementations of this class must be thread-safe.
+ *
+ * <p>This interface should not be implemented directly for unit testing.
+ * Instead, use a mocking framework, such as Mockito, a
```

Comment:
How about this alternate text with `because' and lower MOckito:

Avoid implementing this interface in adaptor unit tests because new methods may
be added in the future. Instead use {@link UnsupportedAdaptorContext}, or use
an automated mock generator like Mockito or {@code java.lang.reflect.Proxy}.